pkg/controllers/scheduler/scheduler.go

```diff
@@ -361,6 +361,39 @@ func (s *Scheduler) prepareToSchedule(
 ) {
 	logger := klog.FromContext(ctx)
 
+	// check migration config
+
+	existingMigrationConfigBytes, migrationConfigExists := fedObject.GetAnnotations()[common.MigrationConfigurationAnnotation]
+	_, customMigrationInfoExists := fedObject.GetAnnotations()[common.AppliedMigrationConfigurationAnnotation]
+	if !migrationConfigExists {
+		if customMigrationInfoExists {
+			delete(fedObject.GetAnnotations(), common.AppliedMigrationConfigurationAnnotation)
+		}
+	}
+	if migrationConfigExists {
+		customMigrationInfoBytes, err := getCustomMigrationInfoBytes(existingMigrationConfigBytes, fedObject, ftc)
+		if err == nil {
+			fedObject.GetAnnotations()[common.AppliedMigrationConfigurationAnnotation] = customMigrationInfoBytes
```

Comment:
Apart from display purposes, what else is `AppliedMigrationConfigurationAnnotation` used for? Should it also be written if the `migrationconfig` and `rsp` plugins are disabled?

Reply:
`AppliedMigrationConfigurationAnnotation` will be used by the scheduler and evict instances through the rsp plugin. For the second question, my answer is yes. This is similiar to auto-migration. Auto-migration will also add an annotation on fedObj which indicates the estimatedCapacity whatever rsp plugin is enabled and disabled. When rsp plugin is disabled, migration does not take effect but annotation will normally be added.

Reply:
AppliedMigrationConfigurationAnnotation will be used by the scheduler and evict instances through the rsp plugin. For the second question, my answer is yes. This is similiar to auto-migration. Auto-migration will also add an annotation on fedObj which indicates the estimatedCapacity whatever rsp plugin is enabled and disabled. When rsp plugin is disabled, migration does not take effect but annotation will normally be added.

Reply:
Why is `MigrationConfigurationAnnotation` insufficient on its own?

Reply:
One main reason is to show the user the currently applied migration config. For example, user may enter a wrong config and scheduler still use the latest migration config. Now user cannot find the applied migration config Intuitively(maybe can be found in events or logs, but it's not convenient). Another one is for user input, we would prefer it to be structured and clearer, while for internal use, the main consideration is the convenience of the program.